### ISSUE TITLE
Final Report: Development of a flexible parametric hurricane model for coastal coupled applications, NOAA Technical Memorandum, 2021.

### ISSUE THREAD

**Issue by saeed-moghimi-noaa** [noaa-ocs-modeling/PaHM#7]

**Comment by pvelissariou1** [comment]
Started the documentation for PaHM.

**Comment by saeed-moghimi-noaa** [comment]
Hi @pvelissariou1  Please update here the state of tech report progress. Thanks, -Saeed

**Comment by pvelissariou1** [comment]
It is a couple of weeks behind. Scheduled for completion by September 27. Preliminary material in place (most handwritten). First draft available by September 20.

**Comment by saeed-moghimi-noaa** [comment]
@pvelissariou1  Please update this issue following our meeting today.  -Saeed

**Comment by pvelissariou1** [comment]
Documentation files have been updates in PaHM.
The updates include:
1) Doxygen documented fortran source codes
2) Doxygen and CMake config files in the doc directory (tested Doxyfile, CMakeLists.txt config files)
3) Created images/*.png files relevant to PaHM documentation (full set of files are kept locally)
4) Created style files for LaTeX and Doxygen in doc/
5) Generated the PaHM html version of the manual as well as a PDF version of it
6) Html/LaTeX documentation is fully hyperlinked and contains "call/called by" hyperlinked relational graphs
Issues:
1) LaTeX does not ouput the sequence of file contents properly (need to reconfigure how doxygen rearranges the doc pages"
2) Resulting doc will be about 500 pages long. LaTeX now produces "compacted" pages as much as possible, thus reducing the documentation for about 80 pages.

**Comment by pvelissariou1** [comment]
The PaHM documentation is part of the PaHM/CoastalApp repositories and it can be produced
from within PaHM using doxygen.
Requirements:
1) Doxygen
2) Latex distribution
3) MathJax (to render the math equations properly, thus producing better visual results)

**Comment by pvelissariou1** [comment]
Added the Doxygen Layout file for html/latex tree generation and display. Needs improvement for more pleasing and organized outputs.

**Comment by saeed-moghimi-noaa** [comment]
@pvelissariou1  
Finalize the 1st draft before 18 Aug 2022.

**Comment by saeed-moghimi-noaa** [comment]
@pvelissariou1
Finalize the 1st draft before 29 Aug 2022.

**Comment by saeed-moghimi-noaa** [comment]
Switched with Parmetis tasks due to need from developers. 

The report will be wrapped up this week.

**Comment by saeed-moghimi-noaa** [comment]
@pvelissariou1 will send the report for review tomorrow (9/9/2022).

**Comment by saeed-moghimi-noaa** [comment]
@pvelissariou1 
- The decision is to wrap up the report for PaHM + ADCIRC coupling through NUOPC/ESMF
- Still some pages need to be added to the report on introduction
- Some pages on model application
- - stand alone
- - coupled
- Remove the pahm code
- add few starting pages for NOAA report 

#### Making sure the PDF is 508 compliance after the review to be submitted to NOAA library
- PPT file is sent to Takis

**Comment by saeed-moghimi-noaa** [comment]
@pvelissariou1 

The report will be submitted by Sep 16, 2022. The report for PaHM + ADCIRC coupling through NUOPC/ESMF.

**Comment by pvelissariou1** [comment]
@saeed-moghimi-noaa

Update:
1) Added most of the additional pages.
2) Temporarily removed the PaHM code part
3) Fixed most of the issues in doxygen generated latex files
    (created scripts to post-process the doxygen generated latex and html files)
4) Doxygen does not support auto-numbering and proper cross reference for figures and sections/subsections
    in the document. Work around for that was to create global definitions for the figure/section links. The cross-references
    look now like:
![cross-ref](https://user-images.githubusercontent.com/63306175/190689614-79735333-8578-4945-9369-cd3ad2686d38.png)
I hope this is acceptable.
5) Working to finalizing report to submit first draft (both in pdf and html)
6) After the final version there will be a post-processing filtering of the latex files so the final report
   will look as expected (tables, sections, figures, etc ...)

**Comment by saeed-moghimi-noaa** [comment]
Thanks for the update @pvelissariou1 , looking forward to the completed 1st draft of the report. -Saeed

**Comment by saeed-moghimi-noaa** [comment]
Update from Monday 9/19/2022
- Work on 1st draft is still on-going.

@pvelissariou1 
- Only focus on finalizing the content in doxygen side (add texts needed for conclusion and elsewhere in the report)
- Generate the latex file
- Work on Latex to finalize the 1st draft

**The continuation of this item is very concerning.**

**Comment by pvelissariou1** [comment]
@saeed-moghimi-noaa 

First draft of the PaHM documentation is online at:
HTML version: [https://noaa-ocs-modeling.github.io/PaHM/pahm_manual.pdf](https://noaa-ocs-modeling.github.io/PaHM/html/index.html)
PDF version:   [https://noaa-ocs-modeling.github.io/PaHM/pahm_manual.pdf](https://noaa-ocs-modeling.github.io/PaHM/pahm_manual.pdf)

**Comment by saeed-moghimi-noaa** [comment]
@pvelissariou1 
Report need to be added. Today Takis will send 1st draft including:  
- Coupled model configuration
- Three 1st pages

**Comment by pvelissariou1** [comment]
@saeed-moghimi-noaa 

The issues with the generated pdf pages have been resolved.
Steps:
1) Create a top level latex file (pahm_manual.tex) that replaces the one created by Doxygen (refman.tex)
    The new file has refined latex commands that at the end produce a more proper pdf document.
2) Allow Doxygen to generate all individual latex files as before
3) Input into pahm_manual.tex all the relevant files generated by Doxygen (step 2)
4) Run pdflatex to generate the final pdf file

The above steps are automated into the Makefile that creates the documentation.
Example of generated file (without the first pages for the moment): [Updated PaHM Manual](https://github.com/noaa-ocs-modeling/PaHM/blob/gh-pages/pahm_manual.pdf)

**Comment by saeed-moghimi-noaa** [comment]
In Review

**Comment by janahaddad** [comment]
@saeed-moghimi-noaa & @pvelissariou1 is this report still under review?

**Comment by pvelissariou1** [comment]
Yes, we left as is. I also need to update the doc/manual. I'll keep it open for the time being.

**Comment by saeed-moghimi-noaa** [comment]
@pvelissariou1 

Please reshape it into the official NOAA reports to get the review process started. See here an example:
[On-demand Automated Storm Surge Modeling Including
Inland Hydrology Effects](https://repository.library.noaa.gov/view/noaa/47926/noaa_47926_DS1.pdf)
Thanks
Thanks,
-Saeed

**Comment by pvelissariou1** [comment]
@saeed-moghimi-noaa Saeed, I have done that and sent it to Zizang as I remember. I can do it again though.

**Comment by saeed-moghimi-noaa** [comment]
Would you please forward that email to me once again?

Thanks

**Comment by pvelissariou1** [comment]
@saeed-moghimi-noaa I remembered now, I sent an email to Ed and you about the report. Anyway, I found the report:
[pahm_report.pdf](https://github.com/noaa-ocs-modeling/PaHM/files/14285561/pahm_report.pdf)
